pkg/reconciler/taskrun/taskrun.go

```diff
@@ -478,6 +478,10 @@ func (c *Reconciler) handlePodCreationError(ctx context.Context, tr *v1beta1.Tas
 		// return a transient error, so that the key is requeued
 		return err
 	}
+	if isTaskRunValidationFailed(err) {
```

Comment:
This series of `if`s might be more clearly written as a switch/case to show that the cases are exclusive?

```go
switch {
case isExceededResourceQuotaError(err):
	backoff, currentlyBackingOff := c.timeoutHandler.GetBackoff(tr.GetNamespacedName(), *tr.Status.StartTime, *tr.Spec.Timeout)
	if !currentlyBackingOff {
		go c.timeoutHandler.SetTimer(tr.GetNamespacedName(), time.Until(backoff.NextAttempt))
	}
	msg = fmt.Sprintf("TaskRun Pod exceeded available resources, reattempted %d times", backoff.NumAttempts)
	tr.Status.SetCondition(&apis.Condition{
		Type:    apis.ConditionSucceeded,
		Status:  corev1.ConditionUnknown,
		Reason:  podconvert.ReasonExceededResourceQuota,
		Message: fmt.Sprintf("%s: %v", msg, err),
	})
case isTaskRunValidationFailed(err):
	tr.Status.MarkResourceFailed(podconvert.ReasonFailedValidation, err)
default:
	// The pod creation failed with unknown reason. The most likely
	// reason is that something is wrong with the spec of the Task, that we could
	// not check with validation before - i.e. pod template fields
	msg = fmt.Sprintf("failed to create task run pod %q: %v. Maybe ", tr.Name, err)
	if tr.Spec.TaskRef != nil {
		msg += fmt.Sprintf("missing or invalid Task %s/%s", tr.Namespace, tr.Spec.TaskRef.Name)
	} else {
		msg += "invalid TaskSpec"
	}
	err = controller.NewPermanentError(errors.New(msg))
	tr.Status.MarkResourceFailed(podconvert.ReasonCouldntGetTask, err)
}
return err
```